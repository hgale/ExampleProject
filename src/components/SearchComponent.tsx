import * as React from 'react';
import { FlatList, KeyboardAvoidingView, StyleSheet, TextInput, View} from 'react-native';
import LoadingIndictor from 'src/components/LoadingIndicator';
import PlaceComponent from 'src/components/PlaceComponent';
import { colors } from 'src/utils/styles';

interface Props {
  loading: boolean;
  results: object[];
  handleInputChange?: (value: string) => void;
}

export default class SearchComponent extends React.Component<Props> {
  private textInput: TextInput | null = null;

  keyExtractor = (item: any) => item.id;

  renderPlace = (rowData: any) => {
    const { item } = rowData;
    return (
      <PlaceComponent description={ item.description } />
    );
  }

  renderSeparator = () => {
    return ( <View style={ styles.line } /> );
  }

  renderPlaces = (loading: boolean, results: object[]) => {
    if (loading) {
      return (<LoadingIndictor style={ styles.LoadingIndicator } /> );
    }
    return (
        <FlatList
          data={ results }
          renderItem={ this.renderPlace }
          ItemSeparatorComponent={ this.renderSeparator }
          keyExtractor={ this.keyExtractor }
        />
    );
  }

  render() {
    const { loading, results } = this.props;
    return (
        <KeyboardAvoidingView style={ styles.container } behavior='padding'>
            <TextInput
                ref={ ref => (this.textInput = ref) }
                placeholder={ 'search for locations' }
                style={ styles.input }
                onChangeText={ this.props.handleInputChange }
            />
            { this.renderPlaces(loading, results) }
        </KeyboardAvoidingView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    backgroundColor: colors.BACKGROUND,
    flex: 1,
    width: '100%',
    justifyContent: 'space-between',
  },
  LoadingIndicator: {
    height:'70%',
  },
  line: {
    height: 1,
    width: '100%',
    backgroundColor: colors.WHITE,
  },
  searchContainer: {
    width: '80%',
    height: 40,
    backgroundColor: 'red',
  },
  input: {
    alignSelf: 'center',
    paddingLeft: 20,
    paddingRight: 20,
    paddingTop: 15,
    paddingBottom: 15,
    borderRadius: 3,
    borderWidth: 1,
    backgroundColor: colors.WHITE,
    borderColor: colors.BLACK,
    margin: 10,
    width: '85%',
  },
});
