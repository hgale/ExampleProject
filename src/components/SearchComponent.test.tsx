import * as React from 'react';
import 'react-native';
import * as renderer from 'react-test-renderer';

import SearchComponent from 'src/components/SearchComponent';

const inputChange = (test : string) => {};

const mockResults = [ {description: 'House of Harry'} ];

it('renders correctly with loading true', () => {
  const component = renderer.create(
    <SearchComponent 
      loading={ true }
      handleInputChange={ inputChange }
      results={ mockResults }
    />);
  expect(component).toMatchSnapshot();
});

it('renders correctly with loading false', () => {
  const component = renderer.create(
    <SearchComponent 
      loading={ true }
      handleInputChange={ inputChange }
      results={ mockResults }
    />);
  expect(component).toMatchSnapshot();
});
