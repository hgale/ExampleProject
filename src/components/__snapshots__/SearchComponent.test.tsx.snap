// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with loading false 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(55, 57, 62)",
        "flex": 1,
        "justifyContent": "space-between",
        "width": "100%",
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="search for locations"
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "rgb(255, 255, 255)",
        "borderColor": "rgb(0, 0, 0)",
        "borderRadius": 3,
        "borderWidth": 1,
        "margin": 10,
        "paddingBottom": 15,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 15,
        "width": "85%",
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(55, 57, 62)",
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="1B9ECC"
      hidesWhenStopped={true}
      size="large"
      style={
        Array [
          Object {
            "height": "70%",
          },
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "padding": 8,
          },
        ]
      }
    />
    <Text
      style={
        Object {
          "color": "rgb(255, 255, 255)",
        }
      }
    >
      Loading...
    </Text>
  </View>
</View>
`;

exports[`renders correctly with loading true 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(55, 57, 62)",
        "flex": 1,
        "justifyContent": "space-between",
        "width": "100%",
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="search for locations"
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "rgb(255, 255, 255)",
        "borderColor": "rgb(0, 0, 0)",
        "borderRadius": 3,
        "borderWidth": 1,
        "margin": 10,
        "paddingBottom": 15,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 15,
        "width": "85%",
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(55, 57, 62)",
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="1B9ECC"
      hidesWhenStopped={true}
      size="large"
      style={
        Array [
          Object {
            "height": "70%",
          },
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "padding": 8,
          },
        ]
      }
    />
    <Text
      style={
        Object {
          "color": "rgb(255, 255, 255)",
        }
      }
    >
      Loading...
    </Text>
  </View>
</View>
`;
